start_esp/nginx-auto.conf.template

```diff
@@ -75,6 +75,26 @@ http {
   server {
     server_name ${ingress.host};
 
+% if experimental_enable_security_config_from_cipherli_st:
+    # The Nginx security configuration recommended on https://cipherli.st/
+    ssl_protocols TLSv1.3; # Requires nginx >= 1.13.0 else use TLSv1.2
+    ssl_prefer_server_ciphers on;
+    ssl_dhparam /etc/nginx/dhparam.pem; # openssl dhparam -out /etc/nginx/dhparam.pem 4096
+    ssl_ciphers ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384;
+    ssl_ecdh_curve secp384r1; # Requires nginx >= 1.1.0
+    ssl_session_timeout  10m;
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_tickets off; # Requires nginx >= 1.5.9
+    ssl_stapling on; # Requires nginx >= 1.3.7
+    ssl_stapling_verify on; # Requires nginx => 1.3.7
+    resolver $DNS-IP-1 $DNS-IP-2 valid=300s;
```

Comment:
we may need to create a GKE e2e test to test it out to make sure at least ESP can start.

Reply:
BTW,  who defines $DNS-IP-1?

Reply:
DNS-IP-1 and DNS-IP-2 should be passed as env var when starting the esp